### PR TITLE
[SPARK-10623][SQL]: fix the predicate pushdown construction

### PR DESCRIPTION
**Pull request by zhzhan**
The predicate pushdown is not working because the construction is wrong. Fix it with startAnd/end

### CODE REVIEW

**Comment by zhzhan** [comment]
@liancheng  @marmbrus  Can you help to review it?

**Comment by liancheng** [comment]
@zhzhan I just opened #8799, which is based on your PR. The reasons are:

1. I'd like to let `buildSearchArgument` to handle all the `startAnd()`/`end()` wrapping stuff.
1. The `in` filter wasn't correctly implemented either.
1. Add more comprehensive tests.
1. `buildSearchArgument` can be further simplified for better readability.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcTest.scala [hunk 1]
```diff
@@ -42,6 +42,12 @@ private[sql] trait OrcTest extends SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  protected def withPPD(f: () => Unit) {
+    sqlContext.setConf("spark.sql.orc.filterPushdown", "true")
+    f()
+    sqlContext.setConf("spark.sql.orc.filterPushdown", "false")
+  }
```
Comment (liancheng) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcTest.scala]: You can use `SQLTestUtils.withSQLConf` instead.

**Comment by zhzhan** [comment]
@liancheng  Thanks for review. Since https://github.com/apache/spark/pull/8799 is opened, which also fix another issue. I will close this one.

**Comment by liancheng** [comment]
@zhzhan Thanks for investigating and working on this issue. Would you mind to help reviewing #8799?

**Pull request closed without merging**